imap_processing/ena_maps/ena_maps.py

```diff
@@ -340,6 +339,97 @@ def __init__(
         self.az_bin_edges = self.sky_grid.az_bin_edges
         self.el_bin_edges = self.sky_grid.el_bin_edges
 
+
+class UltraPointingSet(PointingSet):
+    """
+    Pointing set object specifically for Healpix-tiled ULTRA data, nominally at Level1C.
+
+    Parameters
+    ----------
+    l1c_dataset : xr.Dataset | pathlib.Path | str
+        L1c xarray dataset containing the pointing set data or the path to the dataset.
+        Currently, the dataset is expected to be tiled in a HEALPix tessellation,
+        with data_vars indexed along the coordinates:
+            - 'epoch' : time value (1 value per PSET, from the mean of the PSET)
+            - 'energy_bin_center' : (number of energy bins in L1C)
+            - 'healpix_pixel_index' : HEALPix pixel index
+        Only the 'healpix_pixel_index' coordinate is used in this class for projection.
+    spice_reference_frame : geometry.SpiceFrame
+        The reference Spice frame of the pointing set. Default is IMAP_DPS.
+
+    Raises
+    ------
+    ValueError
+        If the longitude/az or latitude/el bin centers don't match the constructed grid.
+        Or if the longitude or latitude bin spacing is not uniform.
+    ValueError
+        If multiple epochs are found in the dataset.
+    """
+
+    def __init__(
+        self,
+        l1c_dataset: xr.Dataset | pathlib.Path | str,
+        spice_reference_frame: geometry.SpiceFrame = geometry.SpiceFrame.IMAP_DPS,
+    ):
+        # Store the reference frame of the pointing set
+        self.spice_reference_frame = spice_reference_frame
+
+        # Read in the data and store the xarray dataset as data attr
+        if isinstance(l1c_dataset, (str, pathlib.Path)):
+            self.data = load_cdf(pathlib.Path(l1c_dataset))
+        elif isinstance(l1c_dataset, xr.Dataset):
+            self.data = l1c_dataset
+
+        # A PSET must have a single epoch
+        self.epoch = self.data["epoch"].values
+        if len(np.unique(self.epoch)) > 1:
+            raise ValueError("Multiple epochs found in the dataset.")
+
+        # Set the tiling type and number of points
+        self.tiling_type = SkyTilingType.HEALPIX
+        self.num_points = self.data[CoordNames.HEALPIX_INDEX.value].size
+        self.nside = hp.npix_to_nside(self.num_points)
+
+        # Determine if the HEALPix tessellation is nested, default is False
+        self.nested = bool(
+            self.data[CoordNames.HEALPIX_INDEX.value].attrs.get("nested", False)
+        )
+
+        # Get the azimuth and elevation coordinates of the healpix pixel centers (deg)
```

Comment:
In the datasets we use longitude and latitude centers it looks like. Is that what we settled on at the meeting last week? I remember you asked the question about it, but I forgot what the decision was.

Then you use az/el in the classes, but then this `pix2ang` has a `lonlat=True`. So there is a lot of longitude/latitude and azimuth/elevation mixing going on here, including with the CoordNames which you mention below. Would it be feasible to change everything over to longitude/latitude to align with the dataset values so we are consistent everywhere?

Reply:
I am fine leaving that as a follow up PR, but I do really appreciate consistency as you know ;) so yes I do think we should do it if it makes sense.

Reply:
Made a ticket for this in a future PR Issue #1491